client/src/redux/sagas/applicationSaga.js

```diff
@@ -120,10 +121,59 @@ function* getApplicants() {
   }
 }
 
+function* addComment({ payload }) {
+  try {
+    const recruiter = yield select(getCurrentUser)
+    const data = {
+      token: recruiter.token,
+      applicationId: payload.applicationId,
+      data: { comment: payload.comment }
+    }
+
+    const response = yield call(jobApplicationApi.addComment, data)
+
+    if (response.status === 201) {
+      const stages = yield select(getStages)
+
+      const newStages = stages.map(stage => {
+        const commentedApplicant = {
+          ...stage.applicants.find(applicant =>
```

Comment:
Something odd happens here (?)  
even when the find method returns undefined (i.e. stage does not contain the application that got a comment), commentedApplicant is { comments: [{ theNewComment }] }. So this "applicant" is added to all other stages than the one that should be changed (that stage is updated correctly)

Reply:
And the comments are in "applicationComments", not "comments", when fetched. Comments might be better, but...